library/src/hipfft.cpp

```diff
@@ -76,302 +95,168 @@ hipfftResult hipfftPlanMany(hipfftHandle *plan,
+                                           dim, lengths, number_of_transforms, desc));
+            ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_get_work_buffer_size(plan->op_inverse, &workBufferSize));
             break;
         case HIPFFT_Z2Z:
-            if(rocfft_plan_create_internal(plan->ip_forward,
+            ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_create_internal(plan->ip_forward,
                                            rocfft_placement_inplace,
                                            rocfft_transform_type_complex_forward,
                                            rocfft_precision_double,
-                                           1, lengths, number_of_transforms,
-                                           nullptr) != rocfft_status_success)
-                return HIPFFT_INVALID_VALUE;
-            if(rocfft_plan_create_internal(plan->op_forward,
+                                           dim, lengths, number_of_transforms, desc));
+            ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_create_internal(plan->op_forward,
                                            rocfft_placement_notinplace,
                                            rocfft_transform_type_complex_forward,
                                            rocfft_precision_double,
-                                           1, lengths, number_of_transforms,
-                                           nullptr) != rocfft_status_success)
-                return HIPFFT_INVALID_VALUE;
-            if(rocfft_plan_create_internal(plan->ip_inverse,
+                                           dim, lengths, number_of_transforms, desc));
+            ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_create_internal(plan->ip_inverse,
                                            rocfft_placement_inplace,
                                            rocfft_transform_type_complex_inverse,
                                            rocfft_precision_double,
-                                           1, lengths, number_of_transforms,
-                                           nullptr) != rocfft_status_success)
-                return HIPFFT_INVALID_VALUE;
-            if(rocfft_plan_create_internal(plan->op_inverse,
+                                           dim, lengths, number_of_transforms, desc));
+            ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_create_internal(plan->op_inverse,
                                            rocfft_placement_notinplace,
                                            rocfft_transform_type_complex_inverse,
                                            rocfft_precision_double,
-                                           1, lengths, number_of_transforms,
-                                           nullptr) != rocfft_status_success)
-                return HIPFFT_INVALID_VALUE;
+                                           dim, lengths, number_of_transforms, desc));
             break;
         default:
             assert(false);
     }
 
```

Comment:
I would drop the two calls to `rocfft_execution_info_set_work_buffer` above and instead add something like the following here. I didn't find any guarantees that complex-to-complex transforms need no work buffer or that forward and backward transforms need buffers of the same size, so the safest thing is to check all four plans.
```suggestion
 
    size_t workBufferSize2;
    if(plan->ip_forward)
    {
        ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_get_work_buffer_size(plan->ip_forward, &workBufferSize2));
        &workBufferSize = std::max(&workBufferSize2, &workBufferSize);
    }
    if(plan->op_forward)
    {
        ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_get_work_buffer_size(plan->op_forward, &workBufferSize2));
        &workBufferSize = std::max(&workBufferSize2, &workBufferSize);
    }
    if(plan->ip_inverse)
    {
        ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_get_work_buffer_size(plan->ip_inverse, &workBufferSize2));
        &workBufferSize = std::max(&workBufferSize2, &workBufferSize);
    }
    f(plan->op_inverse)
    {
        ROC_FFT_CHECK_INVALID_VALUE(rocfft_plan_get_work_buffer_size(plan->op_inverse, &workBufferSize2));
        &workBufferSize = std::max(&workBufferSize2, &workBufferSize);
    }

```